### PR TITLE
2338 fix

### PR DESCRIPTION
**Pull request by vvarg229**

### CODE REVIEW

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -45,8 +45,19 @@ jobs:
       - name: Checkout neofs-node repository
         uses: actions/checkout@v3
         with:
-          repository: ${{ github.repository }}
-          ref: ${{ github.event.inputs.neofs_node_ref || github.head_ref || github.ref_name }}
+          repository: >
+            ${{
+              github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name ||
+              github.repository
```
Comment (roman-khimov) [inline .github/workflows/run-tests.yml:51]: Have you tried using the `pull/ID/head` trick to keep the same repository?

Comment (vvarg229) [reply .github/workflows/run-tests.yml:51]: > Have you tried using the `pull/ID/head` trick to keep the same repository?

Yes, I tried it.
It's the PRs test results:
https://github.com/vvarg229/neofs-node/pull/4
https://github.com/vvarg229/neofs-node/pull/5

It's the ID/head test result:
https://github.com/vvarg229/neofs-node/actions/runs/4960370107

It's the push to master test result:
https://github.com/vvarg229/neofs-node/actions/runs/4977427703

File: .github/workflows/run-tests.yml [hunk 2]
```diff
@@ -45,8 +45,19 @@ jobs:
       - name: Checkout neofs-node repository
         uses: actions/checkout@v3
         with:
-          repository: ${{ github.repository }}
-          ref: ${{ github.event.inputs.neofs_node_ref || github.head_ref || github.ref_name }}
```
Comment (AnnaShaleva) [inline .github/workflows/run-tests.yml:49]: It's too complicated, why don't you just use the
```
      - name: Checkout neofs-node repository
      - uses: actions/checkout@v3
        with:
          fetch-depth: 0
```
It should wok for both PRs created from forks and from neofs-node.
Do not specify `repository`, do no specify `ref`, the action will do all this magic for you as you described in the commit message. Or not?

Comment (vvarg229) [reply .github/workflows/run-tests.yml:49]: @AnnaShaleva 
`It should wok for both PRs created from forks and from neofs-node.
`
No, it doesn't work.
It's your fix:
https://github.com/vvarg229/neofs-node/blob/master/.github/workflows/run-tests.yml#L50
It's a test PR:
https://github.com/vvarg229/neofs-node/pull/8
It's a result:
https://github.com/vvarg229/neofs-node/actions/runs/5044005505/jobs/9099389504?pr=8
image

Comment (AnnaShaleva) [reply .github/workflows/run-tests.yml:49]: @vvarg229 
> It's your fix: https://github.com/vvarg229/neofs-node/blob/master/.github/workflows/run-tests.yml#L50

But it's not my fix. Are you sure that you've tested without `repository` and `ref` fields set?

Comment (vvarg229) [reply .github/workflows/run-tests.yml:49]: > Are you sure that you've tested without `repository` and `ref` fields set?

Yes, I tested it.
For proof, here is the result:
https://github.com/vvarg229/neofs-node/blob/master/.github/workflows/run-tests.yml#L48
https://github.com/vvarg229/neofs-node/actions/runs/5044005505/jobs/9104414914
image

Comment (roman-khimov) [reply .github/workflows/run-tests.yml]: I think what @AnnaShaleva meant was
```
--- a/.github/workflows/run-tests.yml
+++ b/.github/workflows/run-tests.yml
@@ -45,8 +45,6 @@ jobs:
       - name: Checkout neofs-node repository
         uses: actions/checkout@v3
         with:
-          repository: ${{ github.repository }}
-          ref: ${{ github.event.inputs.neofs_node_ref || github.head_ref || github.ref_name }}
           path: neofs-node
 
 ################################################################
```
Have you tried it this way?

**Review by roman-khimov — COMMENTED** [review]
" workflow: Fix checkout action for PRs from forks " commit message it out of date wrt the change itself.

File: .github/workflows/run-tests.yml (hunk 2 above)
Comment (AnnaShaleva) [reply .github/workflows/run-tests.yml]: Exactly. It should work properly, because it works in neo-go.

**Review by AnnaShaleva — COMMENTED** [review]
@vvarg229, currently your tests are failing due to `go: unknown subcommand "mod"`. This caused by too low go version:
```
Run actions/setup-go@v4
Setup go version spec 1.2
Found in cache @ /opt/github-runner/_work/_tool/go/1.2.2/x64
Added go to the path
Setting GOROOT for Go version < 1.9
Successfully set up Go version 1.2
```
And this issue is caused by the way how go version is specified in your workflow file. We need to use single quotes for go version everywhere in this repo, like this:
```
      - name: Set up Go
        uses: actions/setup-go@v4
        with:
          go-version: '1.20'
          cache: true
```

**Comment by vvarg229** [comment]
> @vvarg229, currently your tests are failing due to `go: unknown subcommand "mod"`. This caused by too low go version:

Yes, I know, that will be fixed in nex PR.
Thank you for highlighting this bug

**Comment by vvarg229** [comment]
> " workflow: Fix checkout action for PRs from forks " commit message it out of date wrt the change itself.

Fixed

**Review by AnnaShaleva — COMMENTED** [review]
Otherwise LGTM.

**Review by roman-khimov — APPROVED** [review]

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**